### PR TITLE
Convert the napatech packet source to a plugin

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11646  

This PR contains the changes to distribute and build the Napatech source modules as a "capture plugin". 

Describe changes:
- Converted the Napatech source modules to be suitable for a capture plugin
- User-level upgrade documentation 

Updates:
- Doc update -- use "Napatech" instead of "NAPATECH"
- Revert unnecessary change to decode module name -- `NapatechDecode`.
- Combine like commits (remove unnecessary fixup commit)

### Provide values to any of the below to override the defaults.

- To use an LibHTP, Suricata-Verify or Suricata-Update pull request,
  link to the pull request in the respective `_BRANCH` variable.
- Leave unused overrides blank or remove.

SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

**Comment by jlucovsky** [comment]
@ralpheastwood I've tested this on a system with a Napatech NT40 4x1G. If you have time to test this, it'd be helpful.

File: doc/userguide/upgrade/8.0-napatech-plugin.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+plugin already, for example::
+
+  plugins:
+    - /usr/lib/suricata/napatech.so
+
+If you are upgrading, you will need to add the location of
+``napatech.so`` to the ``plugins`` section of your ``suricata.yaml``
+manually, as shown above.
+
+Existing Napatech command line options and configuration should continue to work.
```
Comment (jufajardini) [inline doc/userguide/upgrade/8.0-napatech-plugin.rst:35]: same here.

File: plugins/napatech/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+### --capture-plugin=napatech
+
+This is the option that tells Suricata to use a plugin for capture, much like
+`--pcap` tells Suricata to use libpcap or `--af-packet` tells Suricata to use
+AF_PACKET. Here we are telling it to look for a loaded plugin of the name
+`napatech` to provide the capture method.
+
+There is another command line option `--capture-plugin-args` to pass arbitrary
+data on the command line a capture plugin, but this plugin does not yet handle
```
Comment (jufajardini) [inline plugins/napatech/README.md:33]: Are we missing something here? Maybe:
```suggestion
data on the command line to a capture plugin, but this plugin does not yet handle
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
The documentation is rather easy to follow - I think it is good for understanding some concepts about Suricata plugins as a whole. Thanks for that!

One minor thing I noticed is that it seems that throughout the sections the noun version `command line` is being used, instead of the adjective `command-line` one. (nitty picky D: )

**Comment by jlucovsky** [comment]
Continued in #11657

**Pull request closed without merging**